kinds/gen.go

```diff
@@ -24,11 +24,19 @@ import (
 )
 
 func main() {
+	all := getAllKinds()
+	genCoreKinds(all)
+	genK8sImpl(all)
```

Comment:
this is the bit to remove if possible - i couldn't find a spot where we code sniffs the filesystem in the current state of the PR, but AIUI the reason to introduce a second pipeline is to have these jennies that sniff filesystem state and change their behavior. That's the rabbit hole i'd prefer we avoid for as long as reasonably possible - it adds complexity to testing, implementation, and user experience.